lib/inspec/base_cli.rb

```diff
@@ -100,6 +100,23 @@ def diagnose
     end
 
     def opts
+      o = merged_opts
+
+      # Due to limitations in Thor it is not possible to set an argument to be
+      # both optional and its value to be mandatory. E.g. the user supplying
+      # the --password argument is optional and not always required, but
+      # whenever it is used, it requires a value. Handle options that were
+      # defined above and require a value here:
+      %w{password sudo_password}.each do |v|
+        if o[v.to_sym] == -1
+          raise ArgumentError, "Please provide a value for --#{v}. For example: --#{v}=hello."
+        end
+      end
```

Comment:
The trick here is that due to the type constraint these values will never reach `-1`, even if called with `--password=-1` as it's turned into `"-1"`. We can't use `nil` because it just says the default is not defined (or really it's empty, since it cannot check for "not defined"), so this looked like a nice alternative. We could also take the golang approach and pass a `{}`, but since it's a full hash in ruby unlike Golang's empty interface I just picked the next best thing...